arcjet-fastify/index.ts

```diff
@@ -0,0 +1,352 @@
+// our use.
+//
+// Simplify:
+// https://github.com/sindresorhus/type-fest/blob/964466c9d59c711da57a5297ad954c13132a0001/source/simplify.d.ts
+// EmptyObject:
+// https://github.com/sindresorhus/type-fest/blob/b9723d4785f01f8d2487c09ee5871a1f615781aa/source/empty-object.d.ts
+//
+// Licensed: MIT License Copyright (c) Sindre Sorhus <sindresorhus@gmail.com>
+// (https://sindresorhus.com)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions: The above copyright
+// notice and this permission notice shall be included in all copies or
+// substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+// TODO(@wooorm-arcjet): remove.
+declare const emptyObjectSymbol: unique symbol;
+
+// TODO(@wooorm-arcjet): remove.
+type WithoutCustomProps = {
+  [emptyObjectSymbol]?: never;
+};
+
+// TODO(@wooorm-arcjet): remove.
+type PlainObject = {
+  [key: string]: unknown;
+};
+
+/**
+ * Configuration for {@linkcode createRemoteClient}.
+ */
+// TODO(@wooorm-arcjet): this looks like an undocumented feature leaking into the public API.
+// Keep it internal.
+export interface RemoteClientOptions {
+  baseUrl?: string | null | undefined;
+  timeout?: number | null | undefined;
+}
+
+/**
+ * Create a remote client.
+ *
+ * @param options
+ *   Configuration (optional).
+ * @returns
+ *   Client.
+ */
+// TODO(@wooorm-arcjet): this looks like an undocumented feature leaking into the public API.
+// Keep it internal.
+export function createRemoteClient(
+  options?: RemoteClientOptions | null | undefined,
+) {
+  const settings = options ?? {};
+  const baseUrl = settings.baseUrl ?? baseUrlFromEnvironment(process.env);
+
+  return createClient({
+    baseUrl,
+    // @ts-expect-error: TODO(@wooorm-arcjet): register label in protocol.
+    sdkStack: "FASTIFY",
+    sdkVersion: "__ARCJET_SDK_VERSION__",
+    timeout: settings.timeout ?? (isDevelopment(process.env) ? 1000 : 500),
+    transport: createTransport(baseUrl),
+  });
+}
+
+/**
+ * Type that represents a sort-of instance of our `ArcjetFastify` clients:
+ * plain objects that we build on the fly.
```

Comment:
```suggestion
 * Type that represents an instance of the `ArcjetFastify` client.
```
Remove the internal implementation details from the docblock.